File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/QuanHuyenRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface QuanHuyenRow {
    Id?: number;
    TenQuanHuyen?: string;
    TinhThanhId?: number;
    TinhThanhTenTinhThanh?: string;
}

export abstract class QuanHuyenRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'TenQuanHuyen';
    static readonly localTextPrefix = 'SanPhamDB.QuanHuyen';
    static readonly lookupKey = 'SanPhamDB.QuanHuyen';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<QuanHuyenRow>('SanPhamDB.QuanHuyen') }
    static async getLookupAsync() { return getLookupAsync<QuanHuyenRow>('SanPhamDB.QuanHuyen') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<QuanHuyenRow>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/QuanHuyen/QuanHuyenRow.cs
namespace wheystore.SanPhamDB;

[ConnectionKey("Default"), Module("SanPhamDB"), TableName("QuanHuyen")]
[DisplayName("Quan Huyen"), InstanceName("Quan Huyen")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript]
public sealed class QuanHuyenRow : Row<QuanHuyenRow.RowFields>, IIdRow, INameRow
{
    const string jTinhThanh = nameof(jTinhThanh);

    [DisplayName("Id"), Column("ID"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Ten Quan Huyen"), Size(200), NotNull, QuickSearch, NameProperty]
    public string TenQuanHuyen { get => fields.TenQuanHuyen[this]; set => fields.TenQuanHuyen[this] = value; }

    [DisplayName("Tỉnh/Thành"),
     Column("TinhThanhID"),
     ForeignKey(typeof(TinhThanhRow)), LeftJoin(jTinhThanh),
     LookupInclude] 
    public int? TinhThanhId
    {
        get => fields.TinhThanhId[this];
        set => fields.TinhThanhId[this] = value;
    }

    [DisplayName("Tên Tỉnh/Thành"), Origin(jTinhThanh, nameof(TinhThanhRow.TenTinhThanh))]
    public string TinhThanhTen
    {
        get => fields.TinhThanhTen[this];
        set => fields.TinhThanhTen[this] = value;
    }

    public class RowFields : RowFieldsBase
    {

        public Int32Field Id;
        public StringField TenQuanHuyen;
        public Int32Field TinhThanhId;
        public StringField TinhThanhTen;
    }
}
